packages/repository/src/relations/has-many/has-many-through.repository.ts

```diff
@@ -0,0 +1,136 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Count, DataObject, Entity, Filter, Options, Where} from '../..';
+
+/**
+ * CRUD operations for a target repository of a HasManyThrough relation
+ */
+export interface HasManyThroughRepository<
+  Target extends Entity,
+  TargetID,
+  Through extends Entity
+> {
+  /**
+   * Create a target model instance
+   * @param targetModelData - The target model data
+   * @param options - Options for the operation
+   * @returns A promise which resolves to the newly created target model instance
+   */
+  create(
+    targetModelData: DataObject<Target>,
+    options: Options & {
+      throughData?: DataObject<Through>;
+      throughOptions?: Options;
+    },
+  ): Promise<Target>;
+
+  /**
+   * Find target model instance(s)
+   * @param filter - A filter object for where, order, limit, etc.
+   * @param options - Options for the operation
+   * @returns A promise which resolves with the found target instance(s)
+   */
+  find(
+    filter?: Filter<Target>,
+    options?: Options & {
+      throughOptions?: Options;
+    },
+  ): Promise<Target[]>;
+
+  /**
+   * Find a related model by ID
+   * @param targetModelId - The target model ID to find
+   * @param options - Options for the operation
+   * @returns A promise which resolves with the found target instance
+   */
+  findById(
```

Comment:
In a nutshell, we are defining the following methods:

```
create(targetModelData, options);
find(filter, options);
findById(targetModelId, options);
delete(where, options);
deleteById(targetModelId: TargetID, options);
patch(dataObject, where options);
link(targetModelId, options);
unlink(targetModelId, options);
exists(targetModelId, options);
```

As we have discussed when implementing other relations (most notably HasMany), `*ById` methods are not very useful - if you already have the id of the target model, then you can load it directly via the target repository, there is no need to make 2 more queries.

I am proposing to remove `findById` and `deleteById`, at least from the initial implementation. Note that `HasMany` repository does not provide these two methods either.